File: sara-job/web/src/lib/seedProfile.ts
// Real profile data extracted from Akhil's actual resumes (E:\AKHIL\Resumes) —
// used as a one-click seed on the Profile page instead of retyping everything.
// Resume PDFs themselves live in D:\JARVIS\sara-job\resumes (gitignored); actual
// upload/storage wiring comes with the backend, this just records filenames + tracks.

import { Profile } from "./types";

export const SEED_PROFILE: Profile = {
  fullName: "Venkata Akhil Teja Reddy Yamasani",
  email: "yvakhilteja1104@gmail.com",
  phone: "+1 716 709 1439",
  linkedinUrl: "linkedin.com/in/venkata-akhil-teja-reddy",
  githubUrl: "",
  portfolioUrl: "",

  targetRoles: [
    "AI Engineer",
    "Software Engineer",
    "Data Engineer",
    "Machine Learning Engineer",
    "Data Scientist",
  ],
  experienceYears: 3,
  locations: ["New York, NY", "Remote"],
  remoteOk: true,
  salaryMin: 100000,
  salaryMax: 160000,

  // Confirmed directly by Akhil on 2026-07-27 (was previously an unconfirmed guess).
  // Must exactly match one of the <select> options in the Profile form below —
  // an unmatched string silently falls back to showing the first option ("US
  // Citizen") instead of erroring, which is a real trap (looked right, wasn't).
  workAuthorization: "OPT/CPT",
  noticePeriod: "Immediate",

  resumes: [
    { track: "AI Engineer", fileName: "Ai_Engineer_Venkata_Akhil_Teja_Reddy.pdf", notes: "GenAI, RAG & LLM Agents · Python, FastAPI, PostgreSQL" },
    { track: "Software Engineer", fileName: "Software_Engineer_Venkata_Akhil_Teja_Reddy.pdf", notes: "Python Microservices & REST APIs · FastAPI, PostgreSQL, CI/CD" },
    { track: "Data Engineer", fileName: "Data_Engineer_Venkata_Akhil_Teja_Reddy.pdf", notes: "ETL, SQL & Data Modeling · Python, PostgreSQL, AWS" },
    { track: "Machine Learning Engineer", fileName: "Machine_Learning_Venkata_Akhil_Teja_Reddy.pdf", notes: "NLP & Predictive Modeling · Python, MLOps, AWS" },
    { track: "Data Scientist", fileName: "Data_Scientist_Venkata_Akhil_Teja_Reddy.pdf", notes: "Machine Learning, NLP & Statistics · Python, SQL, AWS" },
    { track: "Data Analyst", fileName: "Data_Analyst_Venkata_Akhil_Teja_Reddy.pdf", notes: "SQL, Power BI & Tableau · Python, Dashboards & Insights" },
    { track: "Business Analyst", fileName: "Business_Analyst_Venkata_Akhil_Teja_Reddy.pdf", notes: "Requirements, Analytics & Reporting · SQL, Power BI, Tableau" },
    { track: "QA Analyst", fileName: "QA_Analyst_Venkata_Akhil_Teja_Reddy.pdf", notes: "Quality Assurance & Test Automation · Python, SQL, CI/CD" },
    { track: "Software Tester", fileName: "Software_Tester_Venkata_Akhil_Teja_Reddy.pdf", notes: "API & Data Validation Testing · Python, SQL, CI/CD" },
    { track: "GTM Engineer", fileName: "GTM_Venkata_Akhil_Teja_Reddy.pdf", notes: "AI-Driven Lead Generation & Growth Experiments · Python, Automation, Analytics" },
    { track: "PLM", fileName: "PLM_Venkata_Akhil_Teja_Reddy.pdf", notes: "Configuration Management & PLM · Change management, product data" },
  ],

  screeningAnswers: [
    { question: "Why this company?", answer: "Placeholder — write a real answer per company, or let Sara draft one from the JD once the backend is live." },
    { question: "What is your notice period?", answer: "Immediate." },
    { question: "Are you authorized to work in the US?", answer: "Yes, I am authorized to work in the US (F-1 OPT)." },
    { question: "Will you now or in the future require visa sponsorship?", answer: "No." },
    { question: "Desired salary?", answer: "$100,000–$160,000 depending on role and location." },
  ],

  eeoGender: "Decline to answer",
  eeoRace: "Decline to answer",
  eeoVeteranStatus: "Decline to answer",
  eeoDisabilityStatus: "Decline to answer",

  updatedAt: "",
};

File: sara-job/scripts/seed_profile.py
"""Seed profile_facts / resume_variants / screening_answers from the real
resume data (same source as web/src/lib/seedProfile.ts) so Scout/Fit Scorer
have a real profile to work against before the frontend<->backend sync
(profile currently lives only in browser localStorage) is built.

Safe to re-run — upserts by unique key.

Usage:  python scripts/seed_profile.py
"""

import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parent.parent))

from sqlalchemy.dialects.postgresql import insert

from jarvis.db.models import ProfileFact, ResumeVariant, ScreeningAnswer
from jarvis.db.session import get_session

FACTS = {
    "contact": {
        "full_name": "Venkata Akhil Teja Reddy Yamasani",
        "email": "yvakhilteja1104@gmail.com",
        "phone": "+1 716 709 1439",
        "linkedin_url": "linkedin.com/in/venkata-akhil-teja-reddy",
    },
    "preferences": {
        "target_roles": "AI Engineer, Software Engineer, Data Engineer, Machine Learning Engineer, Data Scientist",
        "locations": "New York NY, Remote",
        "remote_ok": "true",
        "experience_years": "3",
    },
    "compensation": {
        "salary_min": "100000",
        "salary_max": "160000",
    },
    "work_auth": {
        # Guessed from resume timeline (MPS Aug'24-Dec'25, then employed Jan'26) — likely OPT.
        # UNCONFIRMED — do not rely on this for a real application without checking with Akhil.
        "status": "OPT/CPT (unconfirmed — verify before real use)",
        "notice_period": "Immediate",
    },
}

RESUMES = [
    ("AI Engineer", "Ai_Engineer_Venkata_Akhil_Teja_Reddy.pdf", "GenAI, RAG & LLM Agents · Python, FastAPI, PostgreSQL"),
    ("Software Engineer", "Software_Engineer_Venkata_Akhil_Teja_Reddy.pdf", "Python Microservices & REST APIs · FastAPI, PostgreSQL, CI/CD"),
    ("Data Engineer", "Data_Engineer_Venkata_Akhil_Teja_Reddy.pdf", "ETL, SQL & Data Modeling · Python, PostgreSQL, AWS"),
    ("Machine Learning Engineer", "Machine_Learning_Venkata_Akhil_Teja_Reddy.pdf", "NLP & Predictive Modeling · Python, MLOps, AWS"),
    ("Data Scientist", "Data_Scientist_Venkata_Akhil_Teja_Reddy.pdf", "Machine Learning, NLP & Statistics · Python, SQL, AWS"),
    ("Data Analyst", "Data_Analyst_Venkata_Akhil_Teja_Reddy.pdf", "SQL, Power BI & Tableau · Python, Dashboards & Insights"),
    ("Business Analyst", "Business_Analyst_Venkata_Akhil_Teja_Reddy.pdf", "Requirements, Analytics & Reporting · SQL, Power BI, Tableau"),
    ("QA Analyst", "QA_Analyst_Venkata_Akhil_Teja_Reddy.pdf", "Quality Assurance & Test Automation · Python, SQL, CI/CD"),
    ("Software Tester", "Software_Tester_Venkata_Akhil_Teja_Reddy.pdf", "API & Data Validation Testing · Python, SQL, CI/CD"),
    ("GTM Engineer", "GTM_Venkata_Akhil_Teja_Reddy.pdf", "AI-Driven Lead Generation & Growth Experiments · Python, Automation, Analytics"),
    ("PLM", "PLM_Venkata_Akhil_Teja_Reddy.pdf", "Configuration Management & PLM · Change management, product data"),
]

SCREENING_ANSWERS = [
    ("why_this_company", "Why this company?", "Placeholder — write a real answer per company, or let Sara draft one from the JD once that's built."),
    ("notice_period", "What is your notice period?", "Immediate."),
    ("work_authorization", "Are you authorized to work in the US?", "PLACEHOLDER — confirm exact current visa/work-authorization status before this is used on any real application."),
    ("desired_salary", "Desired salary?", "$100,000-$160,000 depending on role and location."),
]


def main() -> None:
    session = get_session()

    for category, kv in FACTS.items():
        for key, value in kv.items():
            stmt = insert(ProfileFact).values(category=category, key=key, value=value)
            stmt = stmt.on_conflict_do_update(
                index_elements=["category", "key"], set_={"value": value}
            )
            session.execute(stmt)

    for track, file_name, notes in RESUMES:
        stmt = insert(ResumeVariant).values(track=track, file_name=file_name, notes=notes)
        stmt = stmt.on_conflict_do_update(
            index_elements=["track"], set_={"file_name": file_name, "notes": notes}
        )
        session.execute(stmt)

    for question_key, question_text, answer in SCREENING_ANSWERS:
        stmt = insert(ScreeningAnswer).values(
            question_key=question_key, question_text=question_text, answer=answer
        )
        stmt = stmt.on_conflict_do_update(
            index_elements=["question_key"], set_={"answer": answer}
        )
        session.execute(stmt)

    session.commit()
    session.close()
    print(f"Seeded {sum(len(v) for v in FACTS.values())} facts, {len(RESUMES)} resume variants, {len(SCREENING_ANSWERS)} screening answers.")


if __name__ == "__main__":
    main()
